packages/editor/src/functions/addMediaNode.ts

```diff
@@ -126,25 +121,54 @@ export async function addMediaNode(
         })
       )
     } else if (contentType.startsWith('model/lookdev')) {
-      const gltfLoader = getState(AssetLoaderState).gltfLoader
-      return await new Promise((resolve) =>
-        gltfLoader.load(url, (gltf) => {
-          const componentJson = gltf.scene.children[0].userData.componentJson
-          EditorControlFunctions.overwriteLookdevObject(
-            [{ name: ModelComponent.jsonID, props: { src: url } }, ...extraComponentJson],
-            componentJson,
-            parent!,
-            before
-          )
-          resolve(null)
-        })
-      )
+      const sceneID = getComponent(getState(EditorState).rootEntity, SourceComponent)
+      const gltfSnapshotStatet = getMutableState(GLTFSnapshotState)
+      const snapshotst = gltfSnapshotStatet[sceneID].snapshots
+      const length = snapshotst.length
+      let reactor
+      reactor = startReactor(() => {
+        const loadingEntity = useHookstate<Entity>(UndefinedEntity)
+        const documentState = useHookstate(getMutableState(GLTFDocumentState))
+        const gltfSnapshotState = getMutableState(GLTFSnapshotState)
+        const snapshots = gltfSnapshotState[sceneID].snapshots
+        useEffect(() => {
+          const spinnerEntity = createLoadingSpinner('lookdev loading spinner', getState(EditorState).rootEntity)
+          console.log('spinner entity', spinnerEntity)
+          loadingEntity.set(spinnerEntity)
+          console.log('adding spinner')
+          const gltfLoader = getState(AssetLoaderState).gltfLoader
+
+          gltfLoader.load(url, (gltf) => {
+            let componentJson = [] as ComponentJsonType[]
+
+            gltf.scene.children.forEach((child) => {
+              componentJson.push(child.userData.componentJson)
+            })
+
+            const mergedComponentJsonArray = componentJson.flat()
+            EditorControlFunctions.overwriteLookdevObject(
+              [{ name: ModelComponent.jsonID, props: { src: url } }, ...extraComponentJson],
+              mergedComponentJsonArray,
+              parent!,
+              before
+            )
+          })
+        }, [])
+        useEffect(() => {
+          if (loadingEntity.value !== UndefinedEntity && snapshots.length !== length) {
+            removeEntityNodeRecursively(loadingEntity.value)
+            loadingEntity.set(UndefinedEntity)
+            if (reactor) reactor.stop()
+          }
+        }, [snapshots.length])
+        return null
+      })
+      return null
```

Comment:
I don't think we need this reactor. We should just keep this simple, add a loading spinner before the gltfloader.load function, and then remove it in the onLoad callback, and add an onError callback that removes it as well.